nomad/leader.go

```diff
@@ -189,6 +189,23 @@ WAIT:
 			goto RECONCILE
 		case member := <-reconcileCh:
 			s.reconcileMember(member)
+		case errCh := <-s.reassertLeaderCh:
```

Comment:
It's not clear to me why leadership fixes are being made as part of this PR. Can you give some context there?

Reply:
Thanks for pointing it out.  Once a snapshot is restored (potentially to represent a previous state of the cluster), we must recreate all leader stateful components (e.g. eval brokers, periodic jobs schedule, etc) as the existing state are no longer valid.

We archive this by having a simulated leadership establishment event, as `establishLeadership` will properly recreate the state.  `operatorSnapshotRestore` triggers such event using the `reassertLeaderCh` in https://github.com/hashicorp/nomad/blob/d13a63c8bbf186de9d18e1c5118cbe3d34af05a6/nomad/operator_endpoint.go#L555-L558 .

I'll add more clarifying comments in that flow.

Reply:
Totally makes sense, thanks!